src/ucp/rndv/proto_rndv.c

```diff
@@ -262,6 +262,9 @@ ucp_proto_rndv_ctrl_init_priv(const ucp_proto_rndv_ctrl_init_params_t *params,
     /* Initialize estimated memory registration map */
     ucp_proto_rndv_ctrl_get_md_map(params, &rpriv->md_map, &rpriv->sys_dev_map,
                                    rpriv->sys_dev_distance);
+    if (params->super.super.rkey_config_key != NULL) {
```

Comment:
Pls add code comment why we're doing this

Reply:
Added

Reply:
or we can do it in the if/else in lines 244-260. Seems we can move the call to ucp_proto_rndv_ctrl_get_md_map before that if/else.